File: java/learning-spring-01/src/main/java/com/example/learningspring/coupled/loosely/game/GameRunner.java
package com.example.learningspring.coupled.loosely.game;

public class GameRunner {
    private final GameConsole game;

    public GameRunner(GameConsole game) {
        this.game = game;
    }

    public void run() {
        System.out.println("Running game: " + game);
    }
}

File: kotlin/learning-spring/src/main/kotlin/com/examplekotlin/learningspring/coupled/loosely/game/GameRunner.kt
package com.examplekotlin.learningspring.coupled.loosely.game

class GameRunner constructor(private var game: GameConsole) {
    fun run() {
        println("Running game: $game")
        game.up()
        game.down()
        game.left()
        game.right()
    }
}